include/pinocchio/spatial/se3-expr-base.hpp

```diff
@@ -0,0 +1,168 @@
+//
+// Copyright (c) 2025 INRIA
+//
+
+#ifndef __pinocchio_spatial_se3_expr_base_hpp__
+#define __pinocchio_spatial_se3_expr_base_hpp__
+
+namespace pinocchio
+{
+  // Forward declaration
+  // TODO: This should go into spatial/fwd.hpp but because of a strange include path
+  // in context/cppadcg.hpp we can include se3.hpp without including spatial/fwd.hpp first.
+  template<typename Derived>
+  struct SE3ExprBase;
+  template<typename Derived>
+  struct SE3ExprNoalias;
+  template<typename RotProduct, typename TransProduct>
+  struct SE3ExprProduct;
+  template<typename RotProduct, typename TransProduct>
+  SE3ExprProduct<RotProduct, TransProduct>
+  make_se3_expr_product(RotProduct rot_prod, TransProduct trans_prod);
```

Comment:
I would suggest a function `make_product` generic, that calls then a template specialization for readability.